python/tank/platform/application.py

```diff
@@ -207,6 +209,17 @@ def log_exception(self, msg):
     ##########################################################################################
     # internal API
 
+    def event(self, event):
```

Comment:
It would be nice to have explicit registration for event types to methods.  This feels like it'll lead to code like

```python
if event_type == "foo":
  do foo
elif event_type == "bar":
  do bar
...
```

as opposed to

```python
# in app init
self.register_handler("file-open", self.on_file_open)
....
```

In that method, which'll wreck havoc with subclassing and documentation.